prov/efa/src/rxr/rxr_pkt_type_req.c

```diff
@@ -2320,9 +2377,21 @@ int rxr_pkt_proc_write_rta(struct rxr_ep *ep, struct rxr_pkt_entry *pkt_entry)
 
 	offset = 0;
 	for (i = 0; i < iov_count; ++i) {
-		ofi_atomic_write_handlers[op][dt](iov[i].iov_base,
-						  data + offset,
-						  iov[i].iov_len / dtsize);
+		/* Get hmem_iface from MR */
+		efa_mr = (struct efa_mr*) ofi_mr_map_get(&ep->util_ep.domain->mr_map, (rta_hdr->rma_iov + i)->key);
+		hmem_iface = efa_mr->peer.iface;
+
+		if (hmem_iface == FI_HMEM_SYSTEM) {
+			ofi_atomic_write_handlers[op][dt](iov[i].iov_base,
+			                                  data + offset,
+			                                  iov[i].iov_len / dtsize);
+		} else {
+			ret = rxr_write_atomic(efa_mr, &iov[i], data + offset, dtsize, op, dt);
+			if (OFI_UNLIKELY(ret != 0)) {
```

Comment:
then here we have `if (OFI_UNLIKELY(err))`

Reply:
done